src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionCheckins/ExposureSubmissionCheckinsViewModel.swift

```diff
@@ -0,0 +1,67 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+class ExposureSubmissionCheckinsViewModel {
+	
+	// MARK: - Init
+	
+	init(checkins: [Checkin]) {
+		self.checkinCellModels = checkins
+			.filter { $0.checkinCompleted } // Only shows completed check-ins
+			.map { ExposureSubmissionCheckinCellModel(checkin: $0) }
+	}
+		
+	// MARK: - Internal
+	
+	enum Section: Int, CaseIterable {
+		case description
+		case checkins
+	}
+	
+	let title = AppStrings.ExposureSubmissionCheckins.title
+	let checkinCellModels: [ExposureSubmissionCheckinCellModel]
+	@OpenCombine.Published var continueEnabled: Bool = false
+	
+	var numberOfSections: Int {
+		Section.allCases.count
+	}
+	
+	var selecteCheckins: [Checkin] {
+		checkinCellModels
+			.filter { $0.selected }
+			.map { $0.checkin }
+	}
+	
+	func numberOfRows(in section: Int) -> Int {
+		switch Section(rawValue: section) {
+		case .description:
+			return 1
+		case .checkins:
+			return checkinCellModels.count
+		case .none:
+			fatalError("Invalid section")
```

Comment:
Log.error and return 0 - might be better than a fatal error?

Reply:
Done